src/components/form/FormElement.tsx

```diff
@@ -1,78 +1,82 @@
 import * as React from 'react';
+import * as classNames from 'classnames/bind';
 
 import * as styles from './FormElement.scss';
 
+const cx = classNames.bind(styles);
+
 export interface FormElementProps {
     name: string;
     helpText?: string;
     errors?: string[];
     showLabel?: boolean;
     required?: boolean;
-    className?: string;
+    __className?: string;
     border?: boolean;
-    case?: boolean;
+    replyError?: boolean;
+    kaseError?: boolean;
+    fieldError?: boolean;
 }
 
-const FormElement: React.SFC<FormElementProps> = (props): JSX.Element => {
-    let errors: JSX.Element[] = [];
-
-    if (props.errors) {
-        props.errors.map((error, idx) => {
-            errors = [
-                ...errors,
-                <div key={idx} className={styles.error}>
-                    {error}
-                </div>
-            ];
-        });
-    }
-
-    let errorDisplay: JSX.Element = null;
+export default class FormElement extends React.PureComponent<FormElementProps> {
+    private getName(): JSX.Element {
+        if (this.props.showLabel && this.props.name) {
+            return <div className={styles.label}>{this.props.name}</div>;
+        }
 
-    if (errors.length > 0) {
-        errorDisplay = (
-            <div
-                className={styles.error}
-                style={{ paddingTop: props.case && 5, paddingLeft: props.case && 10 }}>
-                {errors}
-            </div>
-        );
+        return null;
     }
 
-    let name: JSX.Element = null;
+    private getHelpText(): JSX.Element {
+        if (this.props.helpText && !this.props.errors.length) {
+            return <div className={styles.helpText}>{this.props.helpText} </div>;
+        }
 
-    if (props.showLabel && props.name) {
-        name = <div className={styles.label}>{props.name}</div>;
+        return null;
     }
 
-    let helpText: JSX.Element | string;
+    private getErrors(): JSX.Element {
+        if (this.props.errors.length < 1) {
+            return null;
+        }
 
-    if (props.helpText && !errorDisplay) {
-        helpText = <div className={styles.help_text}>{props.helpText}</div>;
-    } else {
-        helpText = '';
-    }
+        const errors: JSX.Element[] = this.props.errors.map((error, idx) => {
+            const className = cx({
+                [styles.error]: true,
+                [styles.replyError]: this.props.replyError === true,
+                [styles.kaseError]: this.props.kaseError === true,
+                [styles.fieldError]: this.props.fieldError === true
+            });
```

Comment:
What does this produce? Is it a single merged class or somesuch? Just curious what using this library is buying us.

Reply:
Right, exactly it just returns a string containing the classes that will be included. Thought it might be worth introducing to cut down on class concatenation logic in `render()`.